### ISSUE TITLE
BUG: scipy.sparse.linalg.eigsh segmentation fault for complex matrices

### ISSUE THREAD

**Issue by nicolasloizeau** [scipy/scipy#19887]
### Describe your issue.

I get a segmentation fault when a complex matrix is passed to scipy.sparse.linalg.eigsh
scipy version : 1.9.1
numpy version : 1.21.5

### Reproducing Code Example

```python
from scipy.sparse.linalg import eigsh
from scipy.sparse import coo_matrix
import numpy as np

d = 100
np.random.seed(1)
I = np.random.random((d,d))
H = coo_matrix(1j*(I-I.T)) #  random imaginary hermitian matrix
e,v = eigsh(H, k=4, which='SA')
```

### Error message

```shell
Segmentation fault (core dumped)
```

### SciPy/NumPy/Python version and system information

```shell
1.9.1 1.21.5 sys.version_info(major=3, minor=9, micro=12, releaselevel='final', serial=0)
lapack_mkl_info:
  NOT AVAILABLE
openblas_lapack_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
lapack_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
blas_mkl_info:
  NOT AVAILABLE
blis_info:
  NOT AVAILABLE
openblas_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
blas_opt_info:
    libraries = ['openblas', 'openblas']
    library_dirs = ['/usr/local/lib']
    language = c
    define_macros = [('HAVE_CBLAS', None)]
    runtime_library_dirs = ['/usr/local/lib']
```

**Comment by nicolasloizeau** [comment]
> Cannot repro on main
> 
> ```
> In [5]: np.random.seed(12345)
> 
> In [6]: for _ in range(1000):
>    ...:      d = 100
>    ...:      I = np.random.random((d,d))
>    ...:      H = 1j*(I-I.T) #  random imaginary hermitian matrix
>    ...:      e,v = eigsh(H, k=4, which='SA')
> 
> <no segfault, finishes fine>
> ```
> 
> One possibility is that it's been fixed in the meantime. One other possibility though is that there are some matrices which do segfault --- a reproducer with a fixed np.random seed would be helpful.

I've updated the Reproducing Code Example

**Comment by ev-br** [comment]
Thanks. This seems to work fine with the scipy development branch:

````
In [1]: from scipy.sparse.linalg import eigsh
   ...: from scipy.sparse import coo_matrix
   ...: import numpy as np
   ...:
   ...: d = 100
   ...: np.random.seed(1)
   ...: I = np.random.random((d,d))
   ...: H = coo_matrix(1j*(I-I.T)) #  random imaginary hermitian matrix
   ...: e,v = eigsh(H, k=4, which='SA')

In [2]: import scipy

In [3]: scipy.__version__
Out[3]: '1.13.0.dev0+1167.03d16ac'

In [6]: v.shape
Out[6]: (100, 4)

In [7]: e
Out[7]: array([-7.71710393, -7.33242856, -7.08142087, -6.78009764])

```

What I had open at the moment is not exactly scipy main, but something close to it. So would be great to see if you can update scipy to the last released version (1.11) or --- ideally --- the release candidate (https://mail.python.org/archives/list/scipy-dev@python.org/thread/NE65Q5O6CNOTOXS3NOTDIVMFARKPG4LY/) and see if it still crashes on your system.

**Comment by nicolasloizeau** [comment]
right, it worked on 1.11, thanks
